### DISCUSSION TITLE
Checksum error with Symfony UX

### DISCUSSION THREAD

**Discussion by Nek-** [symfony/symfony#57959] [Q&A] [answered]
I have the following message that pops sometimes:

```
Invalid checksum sent when updating the live component.
```

I have no idea how to fix it and do not even have an idea why it pops. For example, in my current workspace when I first click on a checkbox it works (ajax call etc) and when I click a second one it fails with this error... :/
This is not the first time I have this issue and I still have no idea how to fix it or how to learn more about the actual error. 😅  (besides just surrender live components for this use case)

**Comment by WebMamba** [comment]
Could you give us a small reproducer? It's hard to know what's happen here

**Reply by Nek-** [reply]
Sorry not really I used another solution. Next time I will for sure!

_Notice: I think it was related to maybe a missing_ `#[LiveProp]`

**Comment by Nek-** [comment, marked as answer]
I had the error again. So I decided to debug it entirely to help the next person who will experiment it!

# Why and How

The checksum prevents external user data injection while the JS is executed. Therefore it's always generated by the backend in a field named `@checksum`. You can then find this in the attributes of the corresponding live component (as `data-value`).

# Debug it

The best way for me is to put a dump in `LiveComponentHydrator::calculateChecksum()`: `dump($dehydratedPropsData);`.
Then compare the dump you have on error with the last dump you have on the previous request or inside the `data-live-props-value` attribute on your component. You will probably find something wrong with your data!

# Possible causes

- It may be triggered by a `writable: true` missing on a LiveProp
- If you are working with objects and passing through hydrator, serialization, or a way to translate it into a scalar item, then you may have an issue in your transformation process! (maybe a sub-object not transformed?)
